src/components/DownloadImageDialog/ApplicationInstructions.tsx

```diff
@@ -0,0 +1,231 @@
+import has from 'lodash/has';
+import { interpolateMustache } from './utils';
+import { Box, List, Tab, Tabs, Typography } from '@mui/material';
+import { memo, useEffect, useMemo, useState } from 'react';
+import { MUILinkWithTracking } from '../MUILinkWithTracking';
+import {
+	DeviceType,
+	KeysOfUnion,
+	OsSpecificContractInstructions,
+} from './models';
+import { OrderedListItem } from '../OrderedListItem';
+import { Markdown } from '../Markdown';
+
+export type OsOptions = ReturnType<typeof getUserOs>;
+
+export const getUserOs = () => {
+	const platform = window.navigator.platform.toLowerCase();
+	if (platform.includes('win')) {
+		return 'Windows';
+	}
+
+	if (platform.includes('mac')) {
+		return 'MacOS';
+	}
+
+	if (platform.includes('x11') || platform.includes('linux')) {
+		return 'Linux';
+	}
+
+	return 'Unknown';
+};
+
+const dtJsonTocontractOsKeyMap = {
+	windows: 'Windows',
+	osx: 'MacOS',
+	linux: 'Linux',
+} as const;
+
+export const ApplicationInstructions = memo(
+	({
+		deviceType,
+		templateData,
+	}: {
+		deviceType: DeviceType;
+		templateData: { dockerImage: string };
+	}) => {
+		const [currentOs, setCurrentOs] = useState<OsOptions>(getUserOs());
+
+		const instructions = useMemo(() => {
+			if (
+				deviceType?.instructions == null ||
+				Array.isArray(deviceType.instructions) ||
+				typeof deviceType.instructions !== 'object'
+			) {
+				return deviceType?.instructions;
+			}
+
+			const instructionsByOs = deviceType.instructions;
+
+			return Object.fromEntries(
+				(
+					Object.entries(instructionsByOs) as Array<
+						[KeysOfUnion<typeof instructionsByOs>, string[]]
+					>
+				).map(([key, value]) => {
+					const normalizedKey =
+						key in dtJsonTocontractOsKeyMap
+							? dtJsonTocontractOsKeyMap[
+									key as keyof typeof dtJsonTocontractOsKeyMap
+							  ]
+							: (key as keyof OsSpecificContractInstructions);
+					return [normalizedKey, value];
+				}),
+			) as OsSpecificContractInstructions;
+		}, [deviceType?.instructions]);
+		const hasOsSpecificInstructions = !Array.isArray(instructions);
+		const normalizedOs = currentOs === 'Unknown' ? 'Linux' : currentOs;
+
+		useEffect(() => {
+			if (hasOsSpecificInstructions && instructions) {
+				const oses = Object.keys(instructions) as unknown as OsOptions;
+				if (!oses.includes(currentOs) && oses.length > 0) {
+					setCurrentOs(oses[0] as OsOptions);
+				}
+			}
+		}, [currentOs, setCurrentOs, instructions, hasOsSpecificInstructions]);
+
+		if (!deviceType || !instructions) {
+			return (
+				<Typography variant="body1">
+					Instructions for this application are not currently available. Please
```

Comment:
Should it be `application` or `device type`?
```suggestion
					Instructions for this device type are not currently available. Please
```